From 8794979384352010b0672b74995d3bfb8b8048ea Mon Sep 17 00:00:00 2001
From: Jason Andryuk <jandryuk@gmail.com>
Date: Mon, 6 Jun 2022 14:42:08 -0400
Subject: [PATCH] Cleanup pidfile when exiting

The pidfile is left on disk when varstored exits.  Remove it since
varstored is no longer running.

Signed-off-by: Jason Andryuk <jandryuk@gmail.com>
---
 varstored.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

--- a/varstored.c
+++ b/varstored.c
@@ -586,6 +586,7 @@ int
 main(int argc, char **argv)
 {
     struct sigaction sig_handler;
+    char            *pidfile = NULL;
     char            *domain_str;
     char            *ptr;
     int             index;
@@ -651,7 +652,8 @@ main(int argc, char **argv)
             break;
 
         case VARSTORED_OPT_PIDFILE:
-            if (!create_pidfile(optarg))
+            pidfile = optarg;
+            if (!create_pidfile(pidfile))
                 exit(1);
             break;
 
@@ -741,5 +743,8 @@ main(int argc, char **argv)
     if (!db->save())
         return 1;
 
+    if (pidfile)
+        unlink(pidfile);
+
     return 0;
 }
